File: app/src/main/java/org/ametro/ng/ui/navigation/adapter/NavigationCheckBoxItemHolder.java
package org.ametro.ng.ui.navigation.adapter;

import android.view.View;
import android.widget.CheckBox;
import android.widget.TextView;

import org.ametro.ng.R;
import org.ametro.ng.ui.navigation.entities.NavigationCheckBoxItem;
import org.ametro.ng.ui.navigation.entities.NavigationItem;

class NavigationCheckBoxItemHolder implements IHolder
{
    private final CheckBox checkBox;
    private final TextView textView;

    public NavigationCheckBoxItemHolder(View view){
        checkBox = (CheckBox)view.findViewById(R.id.checkbox);
        textView = (TextView)view.findViewById(R.id.text);
    }

    @Override
    public void update(NavigationItem item) {
        NavigationCheckBoxItem checkboxItem = (NavigationCheckBoxItem)item;
        checkBox.setChecked(checkboxItem.isChecked());
        textView.setText(checkboxItem.getText());
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationCheckBoxItemHolder.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter

import android.view.View
import android.widget.CheckBox
import android.widget.TextView
import io.github.romangolovanov.apps.ametro.R
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationCheckBoxItem
import io.github.romangolovanov.apps.ametro.ui.navigation.entities.NavigationItem

internal class NavigationCheckBoxItemHolder(view: View) : IHolder {

    private val checkBox: CheckBox = view.findViewById(R.id.checkbox)
    private val textView: TextView = view.findViewById(R.id.text)

    override fun update(item: NavigationItem) {
        val checkboxItem = item as NavigationCheckBoxItem
        checkBox.isChecked = checkboxItem.checked
        textView.text = checkboxItem.text
    }
}
